src/test/java/org/assertj/vavr/api/MapAssert_containsExactly_Test.java

```diff
@@ -0,0 +1,139 @@
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.error.ShouldNotBeNull.shouldNotBeNull;
+import static org.assertj.vavr.api.VavrAssertions.assertThat;
+
+@SuppressWarnings("unchecked")
+class MapAssert_containsExactly_Test {
+
+    private static final Tuple2<String, String> ENTRY1 = Tuple.of("key1", "value1");
+    private static final Tuple2<String, String> ENTRY2 = Tuple.of("key2", "value2");
+    private static final Tuple2<String, String> ENTRY3 = Tuple.of("key3", "value3");
+
+    @Test
+    void should_pass_if_Map_contains_given_entries_in_order() {
+        final Map<String, String> actual = LinkedHashMap.of(
+                "key1", "value1", "key2", "value2", "key3", "value3"
+        );
+
+        assertThat(actual).containsExactly(ENTRY1, ENTRY2, ENTRY3);
+    }
+
+    @Test
+    void should_pass_if_Map_and_entries_parameter_are_empty() {
+        final Tuple2<String, String>[] entries = new Tuple2[0];
+        assertThat(LinkedHashMap.<String, String>empty()).containsExactly(entries);
+    }
+
+    @Test
+    void should_fail_if_Map_is_not_empty_and_entries_is_an_empty_array() {
+        final Map<String, String> actual = LinkedHashMap.of("key1", "value1", "key3", "value3");
+        final Tuple2<String, String>[] entries = new Tuple2[]{};
+
+        assertThatThrownBy(
+                () -> assertThat(actual).containsExactly(entries)
+        )
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("The array of entries to look for should not be empty");
+    }
+
+    @Test
+    void should_fail_if_entries_parameter_is_null() {
+        assertThatThrownBy(
+                () -> assertThat(LinkedHashMap.<String, String>empty()).containsExactly((Tuple2<String, String>[]) null)
+        )
+                .isInstanceOf(NullPointerException.class)
+                .hasMessage("The array of entries to look for should not be null");
+    }
+
+    @Test
+    void should_fail_if_Map_and_entries_have_different_sizes() {
+        assertThatThrownBy(
+                () -> assertThat(LinkedHashMap.<String, String>empty()).containsExactly(ENTRY1)
+        )
+                .isInstanceOf(AssertionError.class)
+                .hasMessage(
+                        "\nActual and expected should have same size but actual size is:\n" +
+                        " <0>\n" +
+                        "while expected is:\n" +
+                        " <1>\n" +
+                        "Actual was:\n" +
+                        "<LinkedHashMap()>"
+                );
+    }
+
+    @Test
+    void should_fail_if_one_of_entries_is_null() {
+        final Map<String, String> actual = LinkedHashMap.of("key1", "value1", "key3", "value3");
+
+        assertThatThrownBy(
+                () -> assertThat(actual).containsExactly(ENTRY1, null)
+        )
+                .isInstanceOf(NullPointerException.class);
```

Comment:
What about `message` verification here? NPE are sometimes quite hard to debug.

Reply:
True, but this one comes from Vavr directly when converting entries into `LinkedHashMap` and if I'd like to add a message to the exception, I'd have to do something like
```    private <K, V> LinkedHashMap<K, V> asLinkedMap(Tuple2<? extends K, ? extends V>[] entries) {
        return Try.of(() -> LinkedHashMap.ofEntries(entries))
                .mapFailure(
                    Case($(instanceOf(NullPointerException.class)), t -> {
                        if (t.getMessage() == null || t.getMessage().equals("")) {
                            final NullPointerException npe = new NullPointerException("One of entries is null");
                            npe.setStackTrace(t.getStackTrace()); // or npe.initCause(t);
                            return npe;
                        } else {
                            return t;
                        }
                    }))
                .get();
    }```

I don't know is the gain worth of the effort to maintain the code.

Reply:
I see. You can also try something like:
```
if(entries.length != Array.of(entries).filter(Objects::nonNull).length()) {
   // throw meaningful exception here
}
```

Reply:
Yup, that's a nice alternative to check this. Thanks!